### ISSUE TITLE
jboss7.deployed state is too interwined with Artifactory

### ISSUE THREAD

**Issue by kartsm** [saltstack/salt#30515]
The documentation for ``jboss7.deployed`` is dominated by Artifactory-specific information; it should really focus more on JBoss.  Similarly the code has too much Artifactory in it.

Is it really that difficult to write two separate but related states to accomplish the same thing?  This seems like a more [DRY](http://c2.com/cgi/wiki?DontRepeatYourself) implementation strategy.
```yaml
artifact_acquired:
  artifactory.downloaded:
   - artifact:
       artifactory_url: http://artifactory.example.invalid/artifactory
       repository: releases
       artifact_id: webapp
       group_id: invalid.example.webapp
       packaging: war
       version: "3.2.7"
   - target_file: /tmp/webapp.war

application_deployed:
  jboss7.deployed:
    - jboss_config: {{ pillar['jboss'] }}
    - salt_source:
        target_file: /tmp/webapp.war
    - require:
        - artifactory: artifact_acquired
```
Note lack of fetch in ``application_deployed``.  (This might require modification to the Python state code to support.)  That state shouldn't have to care exactly how the deployable artifact arrives, as long as it does arrive.  ``jboss-cli.sh`` doesn't really care either.

Artifactory is a genuinely useful piece of software but there may be other methods of distributing artifacts to one's JBoss nodes.  The ``jboss7.deployed`` state and associated module functions should not be hostile to alternative methods of artifact distribution, nor should they have to be modified every time an alternative is created.

**Comment by jfindlay** [comment]
@kartsm, thanks for the report, your arguments are good, we should decouple the artifactory specificity.

**Comment by kartsm** [comment]
Addressed in PR #30801

**Comment by konradstarzyk** [comment]
Probably it is too late to comment on that, but as the original author of both modules I need to add something here;)
I agree that JBoss state was coupled to Artifactory module, but this was done for a reason.
Fetching the artifact in the separate state before deploying it to Jboss means that you have to download it to the predefined location, for example /tmp/application.war. 
When you are deploying the application you actually want to deploy say version 1.2 version of the application. If you download this file from Artifactory in the separate state you need to download it to /tmp/application.war so that you know the file path and can reference it in the deployment step. This file name does not contain the version information. Then when you deploy this file you cannot simply log into Jboss console and see which version was deployed - the only information you have is that application.war is deployed.
In the previous approach, Jboss state downloaded the file to /tmp/application-1.2.war and then deployed it using Jboss module. Also in the state execution result you could see which version was undeployed and which version was deployed. This information is missing now - you can only see that the application was undeployed and then deployed. You cannot just read the state execution result to see if the deployment version has changed on that server.  Apart from that, now the state has to redeploy the application every time (I cannot be sure here, I think that it was redeployed anyway,  but adding a small fix here would have solved the issue - you could have just deployed the application if its version is different from already deployed one). 
To sum up, decoupling Jboss from Artifactory is probably a good idea, but this fix also removed quite an important feature.
